mvr_bot:
  # 必须有的状态发布器，用于 RL 获取 observation
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 100  # RL通常需要较高的频率，比如 50Hz 或 100Hz

  # --- 下肢 (Legs) ---
  left_hip_pitch_controller:
    type: effort_controllers/JointEffortController
    joint: left_hip_pitch_joint
  left_hip_roll_controller:
    type: effort_controllers/JointEffortController
    joint: left_hip_roll_joint
  left_hip_yaw_controller:
    type: effort_controllers/JointEffortController
    joint: left_hip_yaw_joint
  left_knee_controller:
    type: effort_controllers/JointEffortController
    joint: left_knee_joint
  left_ankle_pitch_controller:
    type: effort_controllers/JointEffortController
    joint: left_ankle_pitch_joint
  left_ankle_roll_controller:
    type: effort_controllers/JointEffortController
    joint: left_ankle_roll_joint

  right_hip_pitch_controller:
    type: effort_controllers/JointEffortController
    joint: right_hip_pitch_joint
  right_hip_roll_controller:
    type: effort_controllers/JointEffortController
    joint: right_hip_roll_joint
  right_hip_yaw_controller:
    type: effort_controllers/JointEffortController
    joint: right_hip_yaw_joint
  right_knee_controller:
    type: effort_controllers/JointEffortController
    joint: right_knee_joint
  right_ankle_pitch_controller:
    type: effort_controllers/JointEffortController
    joint: right_ankle_pitch_joint
  right_ankle_roll_controller:
    type: effort_controllers/JointEffortController
    joint: right_ankle_roll_joint

  # --- 躯干 (Torso) ---
  waist_controller:
    type: effort_controllers/JointEffortController
    joint: waist_joint
  head_controller:
    type: effort_controllers/JointEffortController
    joint: head_joint

  # --- 上肢 (Arms) ---
  left_arm_pitch_higher_controller:
    type: effort_controllers/JointEffortController
    joint: left_arm_pitch_higher_joint
  left_arm_roll_controller:
    type: effort_controllers/JointEffortController
    joint: left_arm_roll_joint
  left_arm_yaw_controller:
    type: effort_controllers/JointEffortController
    joint: left_arm_yaw_joint
  left_arm_pitch_lower_controller:
    type: effort_controllers/JointEffortController
    joint: left_arm_pitch_lower_joint

  right_arm_pitch_higher_controller:
    type: effort_controllers/JointEffortController
    joint: right_arm_pitch_higher_joint
  right_arm_roll_controller:
    type: effort_controllers/JointEffortController
    joint: right_arm_roll_joint
  right_arm_yaw_controller:
    type: effort_controllers/JointEffortController
    joint: right_arm_yaw_joint
  right_arm_pitch_lower_controller:
    type: effort_controllers/JointEffortController
    joint: right_arm_pitch_lower_joint